src/lib.rs

```diff
@@ -941,6 +943,76 @@ pub mod invariant {
         fn get_fee_tiers(&self) -> Vec<FeeTier> {
             self.fee_tiers.get_all()
         }
+
+        #[ink(message)]
+        fn get_liquidity_ticks(&self, pool_key: PoolKey, offset: u16) -> Vec<LiquidityTick> {
+            let mut ticks = vec![];
+            let tick_spacing = pool_key.fee_tier.tick_spacing;
+
+            let max_tick = MAX_TICK - MAX_TICK % tick_spacing as i32;
+            let (chunk_limit, bit_limit) = tick_to_position(max_tick, tick_spacing);
+
+            let mut skipped_ticks = 0;
+
+            for i in 0..=chunk_limit {
+                let chunk = self.tickmap.bitmap.get((i, pool_key)).unwrap_or(0);
+
+                if chunk != 0 {
+                    let end = if chunk as u16 == chunk_limit {
+                        bit_limit
+                    } else {
+                        (CHUNK_SIZE - 1) as u8
+                    };
+
+                    for j in 0..=end {
```

Comment:
IMO this change is more readable: `for j in 0..=end {` -> `for bit in 0..=end {`